esp/src/src-react/components/controls/Grid.tsx

```diff
@@ -0,0 +1,394 @@
+    call: (clearSelection?: boolean) => void;
+    value: number;
+    clear: boolean;
+}
+
+function useRefreshTable(): RefreshTable {
+
+    const [value, setValue] = React.useState(0);
+    const [clear, setClear] = React.useState(false);
+    const call = React.useCallback((clearSelection?: boolean) => {
+        setValue(value + 1);
+        setClear(clearSelection);
+    }, [value]);
+    return { call, value, clear };
+}
+
+export interface FluentStoreStateProps {
+    page: number
+}
+
+export interface FluentStoreStateResponse {
+    selection: any[];
+    setSelection: (selection: any[]) => void;
+    pageNum: number;
+    setPageNum: (pageNum: number) => void;
+    pageSize: number;
+    setPageSize: (pageSize: number) => void;
+    total: number;
+    setTotal: (total: number) => void;
+    refreshTable: RefreshTable
+}
+
+export function useFluentStoreState({ page }: FluentStoreStateProps): FluentStoreStateResponse {
+    const [selection, setSelection] = React.useState([]);
+    const [pageNum, setPageNum] = React.useState(page);
+    const [pageSize, setPageSize] = React.useState(25);
+    const [total, setTotal] = React.useState(0);
+    const refreshTable = useRefreshTable();
+
+    return { selection, setSelection, pageNum, setPageNum, pageSize, setPageSize, total, setTotal, refreshTable };
+}
+
+interface FluentStoreGridProps {
+    store: any,
+    query?: QueryRequest,
+    sort?: QuerySortItem,
+    start: number,
+    count: number,
+    columns: DojoColumns,
+    height: string,
+    refresh: RefreshTable,
+    setSelection: (selection: any[]) => void,
+    setTotal: (total: number) => void,
+}
+
+export const FluentStoreGrid: React.FunctionComponent<FluentStoreGridProps> = ({
+    store,
+    query,
+    sort,
+    start,
+    count,
+    columns,
+    height,
+    refresh,
+    setSelection,
+    setTotal,
+}) => {
+    const memoizedColumns = useDeepMemo(() => columns, [], [columns]);
+    const [sorted, setSorted] = React.useState<QuerySortItem>(sort);
+    const [items, setItems] = React.useState<any[]>([]);
+
+    const selectionHandler = useConst(new Selection({
+        onSelectionChanged: () => {
+            setSelection(selectionHandler.getSelection());
+        }
+    }));
+
+    const refreshTable = useDeepCallback((clearSelection = false) => {
+        if (isNaN(start) || isNaN(count)) return;
+        if (clearSelection) {
+            selectionHandler.setItems([], true);
+        }
+        const storeQuery = store.query({ ...query }, { start, count, sort: sorted ? [sorted] : undefined });
+        storeQuery.total.then(total => {
+            setTotal(total);
+        });
+        storeQuery.then(items => {
+            setItems(items);
+            setSelection(selectionHandler.getSelection());
+        });
+    }, [count, selectionHandler, start, store], [query, sorted]);
+
+    React.useEffect(() => {
+        refreshTable(refresh.clear);
+    }, [refreshTable, refresh.value, refresh.clear]);
```

Comment:
refresh.value is not needed?

Reply:
It is needed - when you click the refresh button on the page, it increments refresh.value under the covers which triggers the refresh - I should probably add a comment.

Reply:
Updated with NOP reference and comment